modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryObjectImpl.java

```diff
@@ -491,6 +492,22 @@ else if (fieldOffLen == BinaryUtils.OFFSET_2)
 
                 break;
 
+            case GridBinaryMarshaller.COMPRESSED: {
+                assert BinaryPrimitives.readByte(arr, fieldPos + 1) == GridBinaryMarshaller.BYTE_ARR;
+
+                int len = BinaryPrimitives.readInt(arr, fieldPos + 1 + 1);
+
+                byte[] compressed = BinaryPrimitives.readByteArray(arr, fieldPos + 1 + 1 + 4, len);
```

Comment:
It will be optimized by compiller, but I think it is more clearly for developer to understand what length consists of.